src/utils/isSameDay.js

```diff
@@ -2,5 +2,9 @@ import moment from 'moment';
 
 export default function isSameDay(a, b) {
   if (!moment.isMoment(a) || !moment.isMoment(b)) return false;
-  return a.isSame(b, 'day');
+  // Compare least significant, most likely to change units first
+  // Moment's isSame clones moment inputs and is a tad slow
```

Comment:
Well I don't think our usecase applies to moment completely because we throw away time. It also seems like given the mutability of moment objects, it's reasonable for them to clone them.

I think this is a good solution for our own usecase.